Source/ablastr/mesh_refinement/coarsen.H

```diff
@@ -34,60 +44,72 @@ namespace CoarsenMR{
      */
     AMREX_GPU_DEVICE
     AMREX_FORCE_INLINE
-    Real Interp ( Array4<Real const> const& arr_src,
-                  GpuArray<int,3> const& sf,
-                  GpuArray<int,3> const& sc,
-                  GpuArray<int,3> const& cr,
-                  const int i,
-                  const int j,
-                  const int k,
-                  const int comp )
+    amrex::Real
+    Interp (
+        amrex::Array4<amrex::Real const> const &arr_src,
+        amrex::GpuArray<int, 3> const &sf,
+        amrex::GpuArray<int, 3> const &sc,
+        amrex::GpuArray<int, 3> const &cr,
+        int const i,
+        int const j,
+        int const k,
+        int const comp
+    )
     {
+        using namespace amrex::literals;
+
+        AMREX_ASSERT(sf[0] == sc[0]);
+        AMREX_ASSERT(sf[1] == sc[1]);
+        AMREX_ASSERT(sf[2] == sc[2]);
+
```

Comment:
Not sure if true, but had we had the impression that is a requirements given line 74.

Reply:
I think this is correct, see my reply to the comment above. However, I wonder if we could move this check on layer up, where `Interp` is called, or something along these lines?

Reply:
Yes, this is pretty deep here - but also only used in Debug mode.

```suggestion
        AMREX_ASSERT_WITH_MESSAGE(sf[0] == sc[0], "Staggering for component 0 does not match!");
        AMREX_ASSERT_WITH_MESSAGE(sf[1] == sc[1], "Staggering for component 1 does not match!");
        AMREX_ASSERT_WITH_MESSAGE(sf[2] == sc[2], "Staggering for component 2 does not match!");

```